src/components/navBar/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+import { useNavigate } from "react-router";
+import NavHome from "@assets/icons/nav_Home.svg?react";
+import NavAll from "@assets/icons/nav_All.svg?react";
+import NavProduct from "@assets/icons/nav_Product.svg?react";
+import NavChat from "@assets/icons/nav_Chat.svg?react";
+import NavMy from "@assets/icons/nav_My.svg?react";
+import NumberBadge from "@components/numberBadge";
+import * as S from "./styles";
+import AlertCountStore from "@stores/AlertCountStore";
+
+const NavBar = () => {
+  const navigate = useNavigate();
+  const [activeButton, setActiveButton] = useState<string>("/");
+  const { chatCount, myPageCount, setChatCount, setMyPageCount } = AlertCountStore();
+
+  const handleButtonClick = (route: string) => {
+    navigate(route);
+    setActiveButton(route);
+  };
+
+  return (
+    <div>
+      <S.BottomNavWrapper>
+        <S.BottomNavButton
+          onClick={() => handleButtonClick("/")}
+          className={activeButton === "/" ? "active" : ""}
+        >
```

Comment:
옹 저는 항상 useLocation으로 현재 경로를 가져와서 설정해줬었는데 useState 이용하는 방법도 있군용 👍🏻👍🏻

Reply:
수빈님이 말씀하신 useLoaction을 보고 든 생각인데, Link 태그와 useLocation을 사용하고 아래 로직을 하나의 컴포넌트로 빼서 반복을 줄일 수도 있을 것 같아요!
```suggestion
        const location = useLocation();
        const isActive = location.pathname === path;

        <Link to={path} className={isActive ? "active" : ""}>
          <Icon />
          {label}
        </Link>
```

Reply:
저 궁금한 점이 하나 있는데, Link 태그를 이용하면 브라우저 상에서 <a>태그로 변환되더라구요. 그런데 어플리케이션 내 페이지 이동은 navigate를 이용하고, 외부 페이지로 이동할 때 <a>태그를 이용하는 것이 좋다고 봤었던 것 같습니다..! 그래서 <a>태그 사용을 지양해야 하지 않을까?하는 생각이 있었는데, 제가 혹시 뭔가 놓친 부분이 있을까요?!

우선 반복되는 코드를 줄일 수 있을 것 같긴해서 말씀해주신 것 참고해서 조금 더 고민해보고 있겠습니다!!!

Reply:
우선 NavButton이라는 컴포넌트를 하나 만들어 반복되는 코드를 줄여줬습니다!
한번 검토 부탁드려요:D~
`src/components/navBar/bottomNavBar/NavButton.tsx`

Reply:
저는 주로 `<Link />`는 클릭시 바로 다른 페이지로 전환될 때 사용, `useNavigation`은 페이지 전환할 때 로직 처리가 필요한 곳에서 사용하곤 했어요..! 
`<Link />`가 브라우저 상에서 태그로 변환되어도 주소만 변경되고 페이지를 새로 고침하는게 아니여서 사용하는 걸로 알고 있습니당!!

Reply:
사용 예시 알려주셔서 감사해요 :D!!
저도 저 질문 남기고 궁금해서 찾아보던 중에 잘 정리된 글을 하나 찾았어서 하나 공유 남기고 갑니당...!
https://velog.io/@ahn-sujin/React-Link-%EC%BB%B4%ED%8F%AC%EB%84%8C%ED%8A%B8%EC%99%80-useNavigate%EC%9D%98-%EC%B0%A8%EC%9D%B4

+) Link는 선언형 , useNavigate는 명령형 이라는 얘기도 있더군용ㅎㅎ
https://stackoverflow.com/questions/71781348/difference-between-link-and-usenavigate-from-react-router-dom

Reply:
솔미님 올려주신 코드 잘 봤습니다!! 저는 Link로 쓰지 못하는 (ex. 핸들러 내부에서 특정 조건을 만족하면 페이지 라우팅을 해야할 때) 곳에서 주로 useNavigate를 사용해 왔었는데요, 기존 html a 태그와 react router의 Link 태그는 솔미님께서 올려주신 블로그 글처럼 작동 방식의 차이가 있기도 하고, useNavigate, handleButtonClick 을 없애는 방향으로 코드를 짤 수도 있을 것 같아서
> 그런데 어플리케이션 내 페이지 이동은 navigate를 이용하고, 외부 페이지로 이동할 때 태그를 이용하는 것이 좋다고 봤었던 것 같습니다..!
해당 내용은 Link로 사용해도 될 것 같다는 의견이었습니다! 고민해주셔서 감사합니다ㅎㅎ